lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -419,6 +421,41 @@ def add_stored_credentials(post, options = {})
         post[:payment_method_options][:card][:mit_exemption][:network_transaction_id] = stored_credential[:network_transaction_id] if stored_credential[:network_transaction_id]
       end
 
+      def add_stored_credential_transaction_type(post, options = {})
+        stored_credential = options[:stored_credential]
+        # Do not add anything unless these are present.
+        return unless stored_credential[:reason_type] && stored_credential[:initiator]
+
+        # Not compatible with off_session parameter.
+        options.delete(:off_session)
+
+        if stored_credential[:initial_transaction]
```

Comment:
Why not instead stored_credential[:initial_transaction] && stored_credential[:initiator] == 'cardholder'?

Reply:
If the first conditional was that then a scenario of `initial_transaction = true and initiator = 'merchant'` would get sent to `else` and marked as `stored_off_session_unscheduled` which is incorrect